### ISSUE TITLE
no-use-before-define when destructuring object with rest operator

### ISSUE THREAD

**Issue by davezuko** [babel/babel-eslint#262]
Hi there. First, just to clarify, I'm not familiar enough with the internals of `eslint` and `babel-eslint` to know where exactly the issue stems from, so if this is not the correct place to report this please just let me know and I'll repost where applicable. Basically, eslint is complaining about variables being used before they're defined when using the rest (spread?) operator while destructuring an object. Here's an example:

**Note**: Destructuring _without_ using the spread operator works as expected with no issues.

## Versions

* `eslint@2.2.0`
* `babel-eslint@5.5.0`

### Failing Cases

Any time the spread operator is used while destructuring the error is thrown (excuse the awful example, it doesn't make much sense for such a simple component -- just for demonstration):

<img width="566" alt="screen shot 2016-02-26 at 6 54 23 pm" src="https://cloud.githubusercontent.com/assets/6439050/13368657/690e2224-dcba-11e5-81a4-1e4d94c18bd7.png">

```js
import React from 'react'

class Todo extends React.Component {
  static propTypes = {
    text: React.PropTypes.string,
  };

  render() {
    // "text" and "rest" are flagged "no-use-before-define"
    const { text, ...rest } = this.props
    
    return <div {...rest}>{text}</div>
  }
}

export default Todo
```

Same error with just the spread operator:

```js
import React from 'react'

class Todo extends React.Component {
  static propTypes = {
    text: React.PropTypes.string,
  };

  render() {
    const { ...rest } = this.props // "rest" is flagged "no-use-before-define"
    
    return <div {...rest}>{this.props.text}</div>
  }
}

export default Todo
```

**Comment by Inkdpixels** [comment]
Thanks @davezuko for reporting this issue! It seems like the problem is already reported in another issue so you might wanna close this one to prevent duplicates. :-) :+1:

https://github.com/babel/babel-eslint/issues/249

**Comment by davezuko** [comment]
Oh perfect, thank you! I will close this when I get home. I did search around first but couldn't find anything, oh well.
